### ISSUE TITLE
a11y: improve input group accessibility

### ISSUE THREAD

**Issue by desig9stein** [IgniteUI/igniteui-angular-samples#2739]
## Description  

Provide an accessible experience for igx-input-group.

1. If the input-group has a label, it should automatically be used as the `aria-label` for the input filed
2. if the input-group has `hint` it should be added to the `aria-describedby` list of the input field
3. `aria-invalid` should be added and automatically updated based on the input's validity state.
4. `aria-required` should be added if the input is a required field. 
5. To notify the user that there is a problem, Add `role="alert"` to the error message.

NOTE:  According to [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/forms/alerts) `aria-required` is required only for browsers that do not yet support ARIA.

 * igniteui-angular version: 11.1.x
 * browser: ALL

## Attachments  
Attach a sample if available, and screenshots, if applicable.

**Comment by Lipata** [comment]
Update current input samples with the requirements above that are applicable.

**Comment by jackofdiamond5** [comment]
@desig9stein, about the first point I'm not sure that we want to explicitly set the `aria-label` of the input if there is an `igxLabel`. The reason being that our input group expects an input with an `igxInput` directive applied to it, so the user could easily set the `aria-label` attribute to said input if they needed it. If we want to set the default value of `aria-label` to be whatever is in the label then we may face some difficulties if we encounter a scenario where a user wants to use a label but does not want to have `aria-label`. That's why I think it's best not to take the control from the user but to leave them to decide whether or not they want an `aria-label` attribute.

About the second point, the hint (`igxHint`) does not have to be something that describes the input. If the hint is for example a notification of sorts, its content may not necessarily describe the input. This means that we may want to leave it up to the user, like the previous point, to decide how they want to add the `aria-describedby` attribute, based on what the hint is actually doing. Is there a specific scenario that you are have in mind?

**Comment by desig9stein** [comment]
> @desig9stein, about the first point I'm not sure that we want to explicitly set the `aria-label` of the input if there is an `igxLabel`. The reason being that our input group expects an input with an `igxInput` directive applied to it, so the user could easily set the `aria-label` attribute to said input if they needed it. If we want to set the default value of `aria-label` to be whatever is in the label then we may face some difficulties if we encounter a scenario where a user wants to use a label but does not want to have `aria-label`. That's why I think it's best not to take the control from the user but to leave them to decide whether or not they want an `aria-label` attribute.
> 
> About the second point, the hint (`igxHint`) does not have to be something that describes the input. If the hint is for example a notification of sorts, its content may not necessarily describe the input. This means that we may want to leave it up to the user, like the previous point, so that they can decide how they want to add the `aria-describedby` attribute, based on what the hint is actually doing. Is there a specific scenario that you have in mind?

@jackofdiamond5 I already discussed this with @Lipata a couple of week ago. I agger with everything that you mention above.

**Comment by jackofdiamond5** [comment]
@desig9stein, alright then we will treat the issue as an enhancement for the input group component. Mainly the part with applying `aria-invalid` when the model is invalid.
